plan/plan.go

```diff
@@ -516,40 +516,83 @@ func (p *Plan) buildEnvs(conf *v2.Config) (map[string]Env, error) {
 	return envPlans, nil
 }
 
-func resolveComponentCommon(commons ...v2.Common) ComponentCommon {
-	providerVersions := copyMap(utilityProviders)
-	var awsPlan *AWSProvider
+var depth = 0
+
+func resolveAWSProvider(commons ...v2.Common) (plan *AWSProvider, providers []AWSProvider, version *string) {
+	depth++
+	if depth > 100 {
+		panic(commons)
+	}
 	awsConfig := v2.ResolveAWSProvider(commons...)
-	additionalProviders := []AWSProvider{}
 	var roleArn *string
-	if awsConfig != nil {
-		if awsConfig.Role != nil {
-			tmp := fmt.Sprintf("arn:aws:iam::%s:role/%s", *awsConfig.AccountID, *awsConfig.Role)
-			roleArn = &tmp
+	// nothing to do
+	if awsConfig == nil {
+		return
+	}
+
+	// set the version
+	version = awsConfig.Version
+
+	// configure the main provider
+	if awsConfig.Role != nil {
+		tmp := fmt.Sprintf("arn:aws:iam::%s:role/%s", *awsConfig.AccountID, *awsConfig.Role)
+		roleArn = &tmp
+	}
+	plan = &AWSProvider{
+		AccountID: *awsConfig.AccountID,
+		Profile:   awsConfig.Profile,
+		Region:    *awsConfig.Region,
+		RoleArn:   roleArn,
+	}
+
+	// grab all aliased regions
+	for _, r := range awsConfig.AdditionalRegions {
+		// we have to take a reference here otherwise it gets overwritten by the loop
+		region := r
+		providers = append(providers,
+			AWSProvider{
+				AccountID: *awsConfig.AccountID,
+				Alias:     &region,
+				Profile:   awsConfig.Profile,
+				Region:    region,
+				RoleArn:   roleArn,
+			})
+	}
+
+	//HACK HACK(el): this is horrible: grab all extra accounts and configure aliased providers for them
+	for aliasPrefix, aws := range awsConfig.AdditionalProviders {
+		// HACK(el): we create this pseudo v2.Common for each additional provider and do the inheritance
+		pseudoCommon := v2.Common{
+			Providers: &v2.Providers{
+				AWS: aws,
+			},
 		}
-		awsPlan = &AWSProvider{
-			AccountID: *awsConfig.AccountID,
-			Profile:   awsConfig.Profile,
-			Region:    *awsConfig.Region,
-			RoleArn:   roleArn,
+		// set so not nil
+		pseudoCommon.Providers.AWS.AdditionalProviders = map[string]*v2.AWSProvider{}
+
+		extraCommons := append(commons[:], pseudoCommon)
+		extraPlan, extraProviders, _ := resolveAWSProvider(extraCommons...)
+		if extraPlan != nil {
+			extraPlan.Alias = util.JoinStrPointers("-", &aliasPrefix, extraPlan.Alias)
+			providers = append(providers, *extraPlan)
 		}
 
+		for _, eP := range extraProviders {
+			extraProvider := eP
+			extraProvider.Alias = util.JoinStrPointers("-", &aliasPrefix, extraProvider.Alias)
```

Comment:
🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Using a reference for the variable on range scope `aliasPrefix` (scopelint)